http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ResourcesRoutes.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package http
+package api
+
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server
+import akka.http.scaladsl.server.Directives._
+import cats.effect.IO
+import cats.mtl.Ask
+import io.opencensus.scala.akka.http.TracingDirective.traceRequestForService
+import org.broadinstitute.dsde.workbench.model.UserInfo
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.broadinstitute.dsde.workbench.leonardo.http.service.ResourcesService
+import org.broadinstitute.dsde.workbench.leonardo.model.BadRequestException
+import org.typelevel.log4cats.StructuredLogger
+
+/**
+ * Routes intended to be used to manages all types of resources handled by leonardo,
+ * which are runtimes and apps
+ */
+
+class ResourcesRoutes(resourcesService: ResourcesService[IO], userInfoDirectives: UserInfoDirectives)(implicit
+  metrics: OpenTelemetryMetrics[IO],
+  logger: StructuredLogger[IO]
+) {
+
+  val routes: server.Route = traceRequestForService(serviceData) { span =>
+    extractAppContext(Some(span)) { implicit ctx =>
+      userInfoDirectives.requireUserInfo { userInfo =>
+        CookieSupport.setTokenCookie(userInfo) {
+          pathPrefix("google" / "v1" / "resources") {
+            pathPrefix(googleProjectSegment) { googleProject =>
+              path("deleteAll") {
+                delete {
+                  parameterMap { params =>
+                    complete(
+                      deleteAllResourcesForGoogleProjectHandler(
+                        userInfo,
+                        googleProject,
+                        params
+                      )
+                    )
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  def deleteAllResourcesForGoogleProjectHandler(userInfo: UserInfo,
+                                                googleProject: GoogleProject,
+                                                params: Map[String, String]
+  )(implicit
+    ev: Ask[IO, AppContext]
+  ): IO[ToResponseMarshallable] =
+    for {
+      ctx <- ev.ask[AppContext]
+      deleteInCloud = params.get("deleteInCloud").exists(_ == "true")
```

Comment:
Right now these would both default to `false` if not specified in the API. Do we want to treat them as required, and throw an exception if they're not present?

Reply:
Oh good catch, I will throw the BadRequest Exception if they are not present, thanks!